### ISSUE TITLE
Milestone cards on project boards

### ISSUE THREAD

**Issue by diegok** [isaacs/github#1375]
I've discovered that it's possible to add other projects cards into a project by pasting the other project URL on a note and I love using it to have a global view of my team's projects. Being able to have the same "hidden" feature for milestones would make my life much easier, as we use milestones for projects having a time limit and ATM I have a note with a link where I need to enter periodically from my board to check progress.

It would be fantastic to have a card showing:

- Name of the milestone (with link)
- Progress bar of completion (like the project card)
- End date and time left before it or overdue time (in red)

**Comment by Reza1024** [comment]
I was going to file the exact same issue 🙂 we can put issues and projects as a card in another project. It would be great to have the ability to put milestones too.

**Comment by lukehefson** [comment]
👋 I work at GitHub and noticed this issue. If anyone is following and interested in this idea – I'd love to know more about your use case(s)!

Specifically:
- Regardless of project boards what do you tend to use Milestones for?
- If you could attach a notion of a due date to a column in a project board would you still need this?

**Comment by diegok** [comment]
Grouping and due dates is what we used to use milestones for. Since we're not able to add those to our projects we are using columns for grouping/prioritizing and several sub-optimal methods for due dates.

Due dates on columns will solve most of our use cases. 

I'm trying to think  a use case for including milestones and there are no strong ones. Being able to add milestones would make sense on one of our management projects as we would be able to see more easily several groups with due dates, but 99.9% of the times we don't need that and having due dates on colums will do it! :+1:

**Comment by chase-moskal** [comment]
would a milestone be better represented as an individual card, or as a whole column which contains issues?

"milestone columns"?  
i'd like to drag issues in and out of milestones on the board

although, i do already like the way the column/project is displayed in each issue's sidebar..  
it just feels a little bit like i'm using project columns *instead of* milestones.. but maybe that's perfectly okay :)

**Comment by smeijer** [comment]
> it just feels a little bit like i'm using project columns instead of milestones.. but maybe that's perfectly okay :)

Yeah, I use a combination of both.

The project is basically just kanban, `backlog | in progress | QA | code review | done`. The milestones are there to prioritize the backlog a bit, without the need for creating a new board for every sprint.

That way we can keep adding tickets to the project, while the developer doesn't see them, as long as he filters on his current milestone. As soon as his milestone runs empty, he can change the milestone filter, and see the new tickets.

This flow allows us to push new tickets to an older milestone if they are blocking a release. If the developer would already have moved on to a different board, they'll miss this kind of tickets.

If this issue will result in "embedding milestones in notes", then there is one more thing I would like to request that I think is within or close to this scope.

We add `issues` to a project, but as soon as the `issue` is `in progress`, there is also a `pull-request` available, and added to the same project. This results in having two cards. It would be nice if those two were grouped somehow. 

Basically, if there is a PR on a board, that has a `closes #x` (or other magic keyword), it could nest the related issue. Instead of having those two show up as individual cards.

**Comment by ricsimps** [comment]
@lukehefson In my group we have multiple dev teams working on the same products. Within each team's sprint they will have 5-6 initiatives they're working on, which all have different delivery dates and product owners.  We use milestones as a way to bucket-ize initiatives.  You look at the milestone and you can see the open issues that are relevant to a release date/product owner.

Normally, teams turn to labels for bucket-izing but because of the sheer number of teams we have, labels are pretty locked down to standard set that all teams can agree on.  Which has a downside in that, as a team you can't really prioritize within a milestone well.

We're interested in using Projects with milestones, so each team could have a column for the milestones they're working which lists the open issues in that milestone.  Also, we could add notes in the column to separate the list further between show-stoppers, fast follow, non-blocker issues.  It could provide that quick summary of where we're at with issues in regards to our release dates.  

It would also be a good manage management tool.  Every time they ask, "How we doing on bugs?"  We could point them to the projects page and get on with our day. No need for them to filter anything

**Comment by efroim102** [comment]
I actually contacted GitHub a while back to ask for this specific feature... I referred to as Card URL parsing/rendering, and wanted Milestone URL's to be parsed similar to the way Project & Issue URL's are parsed/rendered.

**Comment by woodyhayday** [comment]
We'd like to see this too, and ultimately what'd be a winning feature for us would be more granular automations:

Project Column Automations to include milestones, (e.g. `Automatically pull issues assigned to a particular milestone into column`)

... this would save us a step in our flow, where by we prioritise with milestones, then move that bucket into the 'todo' column, and potentially stop issues falling through the cracks on larger projects.

**Comment by sanderson** [comment]
My team works on multiple projects with multiple milestones. It would be nice to be able to create a card for each milestone that displays the milestone progress. We would essentially have a column of milestone cards that gives a good overview of where our overall progress stands.

**Comment by emeka** [comment]
In our company, we use Epics as a higher level view on our initiatives.  We include them in a Kanban and use it for higher level prioritization across the company.  We are doing this in Jira.
If we were able to place milestones in a project board as a card, then we would not need to use Jira :-)

**Comment by lukehefson** [comment]
@jeffrey-kissel Although there isn't a Milestone 'card type' supported on Project boards – you _can_ filter by milestones on a project board.

So if you were to use a Milestone as a major step and then issues as all the tasks needed to complete that step, when all the issues under a given milestone are mixed together with other issues on a board (perhaps because you're using a board to manage other tasks as well) you can click on the little milestone icon in a card to filter by that milestone.

**Comment by timharris777** [comment]
+1 for being able to add a milestone as a card on a project board. Here is the use case:

I have a project board that is my public roadmap. I only want to show the "milestones" on the roadmap, not all the individual issues that make up each milestone. We would have a different board for all the issue manageement.

Basically, we need a higher level board to track all the different milestones and their progress. This is not easily done. As an aside, GitKraken currently supports this.
